### DISCUSSION TITLE
Any fake app can create new users in my project

### DISCUSSION THREAD

**Discussion by cddqssc** [appwrite/appwrite#6874] [Q&A] [answered]
It's very easy for people to obtain the project ID and app package name. Anyone can create new users in my project, and I can't determine which users are created by my app or by other fake apps. This makes me feel extremely frustrated.

I consider this issue to be very serious. Are there any solutions available to address this problem at the moment?

**Comment by stnguyen90** [comment, marked as answer]
@cddqssc, thanks for opening this! Protecting from this type of abuse is a common problem for any app, and any resource you find online can probably be applied to Appwrite. 

Some related feature requests you might want to 👍🏼 are:

* https://github.com/appwrite/appwrite/issues/4791
* https://github.com/appwrite/appwrite/issues/5384

For now, for mobile apps, you can:

1. Disable creating accounts client-side by setting the Users limit to 1 in the Appwrite console > Security > Uses limit.
1. Integrate with [Google's Integrity API](https://developer.android.com/google/play/integrity) or [Apple's App Attest Service](https://developer.apple.com/documentation/devicecheck/establishing_your_app_s_integrity). After doing so, you'll get some sort of code from the respective service
2. Create an Appwrite function that will take details from the sure and the code as input, validate the code against the provider, and then create the account only if validation passes.

**Reply by cddqssc** [reply]
Your answer has helped me, thank you.

**Reply by cddqssc** [reply]
@stnguyen90 Sorry to disturb you, but may I ask another question: For a Flutter app, will the projectID be exposed in API requests? If it won't be exposed, I would like to consider it as a secret key

**Reply by stnguyen90** [reply]
Yes, it will be exposed and that's fine. Nothing you put in a mobile app is safely secret.

**Reply by minecrawler** [reply]
I guess the problem is that once you get rid of appwrite API access, you'll open up access to your custom API, which can be abused just as well. So it needs additional security to pretect it from abuse - which is very hard to do, since you need to distribute the way of accessing it. The above mentioned mobile app alternative and captcha for web seem to be the way, however there's a machinery already in place to break even these if anyone really wants to target you. Will keep the bots out, at least!

Another idea is to create a whitelist for new registration. Just let "@gmx.com" in, so "bwefyu32@vhv.tk" cannot enter. But you'll always get support requests from people with custom domains or from esoteric mail services :/

I'd really love to see appwrite commit to best-practice solutions and pushing them in the docs and the API so it's at least one concern fewer to figure out for developers.
